core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -284,14 +286,30 @@ public Table loadTable(SessionContext context, TableIdentifier identifier) {
             tableFileIO(response.config()),
             response.tableMetadata());
 
-    BaseTable table = new BaseTable(ops, fullTableName(loadedIdent), this.scanReporter);
+    final TableIdentifier tableIdentifier = loadedIdent;
```

Comment:
In Iceberg, we avoid using `final` for local variables because Java is now good at detecting this automatically and it doesn't make much of a difference for performance.